src/framework/preferences/js/SelfVoicingEnactor.js

```diff
@@ -0,0 +1,151 @@
+/*
+Copyright 2013-2015 OCAD University
+
+Licensed under the Educational Community License (ECL), Version 2.0 or the New
+BSD license. You may not use this file except in compliance with one these
+Licenses.
+
+You may obtain a copy of the ECL 2.0 License and BSD License at
+https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
+*/
+
+var fluid_2_0 = fluid_2_0 || {};
+
+(function ($, fluid) {
+    "use strict";
+
+    /*******************************************************************************
+     * speak
+     *
+     * An enactor that is capable of speaking text.
+     * Typically this will be used as a base grade to an enactor that supplies
+     * the text to be spoken.
+     *******************************************************************************/
+
+    fluid.defaults("fluid.prefs.enactor.speakEnactor", {
+        gradeNames: ["fluid.prefs.enactor", "autoInit"],
+        preferenceMap: {
+            "fluid.prefs.speak": {
+                "model.value": "default"
+            }
+        },
+        model: {
+            value: false
+        },
+        events: {
+            onStart: null,
+            onStop: null,
+            onPause: null,
+            onResume: null,
+            onError: null,
+            onTextQueued: null
+        },
+        components: {
+            tts: {
+                type: "fluid.textToSpeech",
+                options: {
+                    listeners: {
+                        "onCreate.clear": "{that}.cancel",
+                        "start.hoist": "{speakEnactor}.events.onStart",
+                        "end.hoist": "{speakEnactor}.events.onStop",
+                        "pause.hoist": "{speakEnactor}.events.onPause",
+                        "resume.hoist": "{speakEnactor}.events.onResume",
+                        "error.hoist": "{speakEnactor}.events.onError"
+                    }
+                }
+            }
+        },
+        invokers: {
+            announce: {
+                funcName: "fluid.prefs.enactor.speakEnactor.announce",
```

Comment:
This should really be named something like "queueAnnounce" or "queueSpeech" and the underlying method in TTS should be given the same name for consistency - to emphasise to the reader that it does not speak the supplied text immediately but adds it to a queue.